onmt/dynamic/transforms/tokenize.py

```diff
@@ -0,0 +1,302 @@
+class ONMTTokenizerTransform(TokenizerTransform):
+    """OpenNMT Tokenizer transform class."""
+
+    def __init__(self, opts):
+        """Initialize neccessary options for OpenNMT Tokenizer."""
+        super().__init__(opts)
+        self._parse_opts()
+
+    @classmethod
+    def add_options(cls, parser):
+        """Avalilable options relate to Subword."""
+        super().add_options(parser)
+        group = parser.add_argument_group('Transform/Subword/ONMTTOK')
+        group.add('-src_subword_type', '--src_subword_type',
+                  type=str, default='none',
+                  choices=['none', 'sentencepiece', 'bpe'],
+                  help="Type of subword model for src (or shared) in onmttok.")
+        group.add('-tgt_subword_type', '--tgt_subword_type',
+                  type=str, default='none',
+                  choices=['none', 'sentencepiece', 'bpe'],
+                  help="Type of subword model for tgt in onmttok.")
+        group.add('-onmttok_kwargs', '--onmttok_kwargs', type=str,
+                  default="{'mode': 'none'}",
+                  help="Accept any OpenNMT Tokenizer's options in dict string,"
+                  "except subword related options listed earlier.")
+
+    def _set_subword_opts(self):
+        """Set all options relate to subword for OpenNMT/Tokenizer."""
+        super()._set_subword_opts()
+        self.src_subword_type = self.opts.src_subword_type
+        self.tgt_subword_type = self.opts.tgt_subword_type
+
+    def _parse_opts(self):
+        self._set_subword_opts()
+        logger.info("Parsed additional kwargs for OpenNMT Tokenizer {}".format(
+            self.opts.onmttok_kwargs))
+        self.other_kwargs = self.opts.onmttok_kwargs
+
+    @classmethod
+    def get_specials(cls, opts):
+        src_specials, tgt_specials = set(), set()
+        if opts.onmttok_kwargs.get("case_markup", False):
+            _case_specials = ['｟mrk_case_modifier_C｠',
+                              '｟mrk_begin_case_region_U｠',
+                              '｟mrk_end_case_region_U｠']
+            src_specials.update(_case_specials)
+            tgt_specials.update(_case_specials)
+        return (set(), set())
+
+    def _get_subword_kwargs(self, side='src'):
+        """Return a dict containing kwargs relate to `side` subwords."""
+        subword_type = self.tgt_subword_type if side == 'tgt' \
+            else self.src_subword_type
+        subword_model = self.tgt_subword_model if side == 'tgt' \
+            else self.src_subword_model
+        kwopts = dict()
+        if subword_type == 'bpe':
+            kwopts['bpe_model_path'] = subword_model
+            kwopts['bpe_dropout'] = self.alpha
+        elif subword_type == 'sentencepiece':
+            kwopts['sp_model_path'] = subword_model
+            kwopts['sp_nbest_size'] = self.subword_nbest
+            kwopts['sp_alpha'] = self.alpha
+        else:
+            raise ValueError(f'Unvalid subword_type: {subword_type}.')
```

Comment:
This case should be handled. OpenNMT Tokenizer does not necessarily need to apply any subword method.

Reply:
Probably some `elif isinstance(subword_type, str)` and no else (so that the empty dict is returned) would do the trick.